compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -399,6 +399,15 @@ object SymDenotations {
     /** The name with which the denoting symbol was created */
     final def originalName(implicit ctx: Context): Name = initial.effectiveName
 
+    /** The owner with which the denoting symbol was created. */
+    final def originalOwner(implicit ctx: Context): Symbol =
+      if (exists) {
+        // There is a similar logic in DottyBackendInterface.originalLexicallyEnclosingClass (see comment there)
+        val validity = initial.validFor
+        val shiftedContext = ctx.withPhase(validity.phaseId)
+        Symbols.toDenot(symbol)(shiftedContext).maybeOwner
```

Comment:
I think we should be able to replace all the logic here by just `initial.maybeOwner`.

Reply:
Yup, that worked :) Much better!